bug: Cannot find name 'AbiParameter'

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.8.3

### Current Behavior

I am encountering this error when building the app after transitioning from web3-react to wagmi. I'm not exactly sure if there are any problems with any dependencies and I'd appreciate if you guys could take a look.

node_modules/@wagmi/core/dist/contracts-95eb7c72.d.ts:143:108 - error TS2552: Cannot find name 'AbiParameter'

### Expected Behavior

To compile without errors in the node_modules.

### Steps To Reproduce

I am using the following dependencies:

Installing and building with those dependencies should be enough to reproduce the issue.

```
"dependencies": {
    "@ant-design/icons": "=4.7.0",
    "@headlessui/react": "=1.4.3",
    "@notifi-network/notifi-react-card": "=0.35.0",
    "@reduxjs/toolkit": "=1.8.1",
    "axios": "=0.25.0",
    "d3": "=7.3.0",
    "ethers": "^5.7.2",
    "events": "=3.3.0",
    "immer": "=9.0.12",
    "moment": "=2.29.4",
    "react": "=17.0.2",
    "react-countup": "=6.2.0",
    "react-datepicker": "=4.7.0",
    "react-device-detect": "=2.1.2",
    "react-dom": "=17.0.2",
    "react-player": "=2.9.0",
    "react-redux": "=7.2.6",
    "react-responsive": "=9.0.0-beta.6",
    "react-router-dom": "=6.2.2",
    "react-spinners": "=0.11.0",
    "react-toastify": "=8.1.1",
    "redux": "=4.1.2",
    "redux-logger": "=3.0.6",
    "redux-persist": "=6.0.0",
    "redux-thunk": "=2.4.1",
    "typechain": "=8.1.1",
    "wagmi": "^0.8.3"
  }

```

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

_No response_

### Anything else?

_No response_

--- a/.changeset/gold-lies-crash.md
+++ b/.changeset/gold-lies-crash.md
@@ -1,0 +1,6 @@
+---
+'@wagmi/core': patch
+'wagmi': patch
+---
+
+Fixed internal type

--- a/packages/core/src/internal.ts
+++ b/packages/core/src/internal.ts
@@ -1,6 +1,7 @@
 export { debounce } from './utils'
 
 export type {
+  AbiParameter,
   ContractConfig,
   ContractsConfig,
   ContractsResult,

--- a/packages/core/src/types/contracts.ts
+++ b/packages/core/src/types/contracts.ts
@@ -2,8 +2,8 @@ import type {
   Abi,
   AbiEvent,
   AbiFunction,
-  AbiParameter,
   AbiParameterToPrimitiveType,
+  AbiParameter as AbiParameter_,
   AbiParametersToPrimitiveTypes,
   AbiStateMutability,
   Address,
@@ -15,6 +15,11 @@ import type {
 import type { ethers } from 'ethers'
 
 import type { IsNever, Join, NotEqual, Or } from './utils'
+
+// TODO: Removed during build for some reason so proxying here
+// and exporting from `src/internal.ts` for now
+// https://github.com/wagmi-dev/wagmi/issues/1306
+export type AbiParameter = AbiParameter_
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // Contract Configuration Types

